### ISSUE TITLE
Bracket bevy feedback

### ISSUE THREAD

**Issue by lecoqjacob** [amethyst/bracket-lib#269]
Hey just wanted to share some feedback while poking around in the bevy branch!

First off, on exiting of the program, bracket-bevy seems to throw an error on unwrapping of the window in the mouse system 

```

let (camera, camera_transform) = q_camera.single();
    let wnd = if let RenderTarget::Window(id) = camera.target {
        wnds.get(id).unwrap()
    } else {
        wnds.get_primary().unwrap()
    };

```

Which is odd since bracket-bevy is just a plug-in of systems running in bevy so I would think bevy wouldn't update the system on app exit. A quick google search didn't lead me down and successive results on why the system would still run.

~~Point 2: it seems that when importing bracket-bevy, and bevy as specific git branches, I am clashing when importing bracket-pathfinding. When implementing base map or alrogithm2D, the compiler is complaining that two different versions of bracket-geometry are being used. Pretty odd since they are locked at the same version.~~

I plan to continue to mess around with the plug-in but would love to help out if you could use it!

EDIT:

Sorry im just dumb on point #2. I have to use the specific git branch for each bracket library. Silly me

Edit #2: 
Did some more digging into the window unwrapping issue when the window is closed manually or by the `bevy::window::close_on_esc`. Seems bevy closes every window, and then once every window is closed, it then sends out the AppExit event which triggers the shutdown of the application. If you simple insert the `AppExit` event, everything shuts down cleanly without the error being produced.

```
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_plugin(BTermBuilder::simple_80x50().with_random_number_generator(true))
        .add_startup_system(setup)
        .add_system(tick)
        .add_system(exit_system)
        .run();
}

fn exit_system(mut exit: EventWriter<AppExit>) {
    exit.send(AppExit);
}
```

**Comment by lecoqjacob** [comment]
One issue with the scaler, for non-resizing modes, the scaler can produce negative usize results resulting in a panic by rust. My debugs are not very helpful, but you can see that `h` (height) is 25, but the `step_pos.1` is 26 resulting in a negative overflow.

Testing with `TerminalScalingMode::ResizeTerminals` works fine since it actually updates the height and width

https://user-images.githubusercontent.com/9278174/173469540-3e16fb00-c5b4-4e36-a028-be0bc35b0051.mp4

**Comment by lecoqjacob** [comment]
Went ahead and fixed the window issue! https://github.com/amethyst/bracket-lib/pull/270

**Comment by thebracket** [comment]
There's a few overflow crashes right now. I goofed when I translated from Bevy and used a bunch of `usize` where `i32` made sense (negatives can make sense when calculating UI positions, even if they aren't used) - as well as a few "print command parameters are in a different order than the main library" type things - so I'll be fixing those. Once that's done, I'll get some safety checks into the scaler.

Funny Bevy issue. I can't find a way to set window size from a plugin, because the window size resource is read before plugin instantiation. :-|
